ApiData.ts

```diff
@@ -0,0 +1,264 @@
+
+		if (to >= this.maxRange) {
+			const lastPage = Math.floor(this.maxRange / this.pageSize) + 1;
+			this.currentPage = lastPage;
+			from = this.maxRange - this.pageSize;
+			to = this.maxRange;
+		}
+
+		return this.fetchAndProcessRecords(from, to)
+			.then(processedData => {
+				this.data = processedData;
+				this.displayRecords();
+				this.updatePageInfo();
+			})
+			.catch(error => {
+				console.error('Failed to fetch records:', error);
+				alert('Error occured while fetching records!');
+			});
+	}
+
+	/** search through records using fromID */
+	searchRecords(searchValue: number): Promise<void> {
+		// Maximum allowed Value
+		const maxRange = this.maxRange
+
+		if (searchValue >= 0 && searchValue <= maxRange) {
+			let from = searchValue;
+			const pageSize = this.pageSize;
+
+			if (from + pageSize > maxRange) {
+				from = Math.max(0, maxRange - pageSize);
+			}
+
+			return this.fetchAndProcessRecords(from, from + pageSize)
+				.then(processedData => {
+					this.data = processedData;
+					this.currentPage = Math.ceil(from / this.pageSize) + 1;
+					// Set firstVal to searched value
+					this.firstVal = from;
+					// Calculate lastVal based on pageSize
+					this.lastVal = from + this.pageSize - 1;
+					this.displayRecords();
+					this.updatePageInfo();
+					// empty search input after searching 
+					$('#fromInput').val('');
+				})
+				.catch(() => {
+					throw ('Failed to search value');
+				});
+		} else {
+			alert(`Error while searching , please enter values in the range (0-${maxRange})`);
+			return Promise.resolve();
+		}
+	}
+
+	/** use Ajax for data fetching */
+	private async fetchData(url: string): Promise<number | string> {
+		$('#overlay').show();
+		const response = await $.ajax({
+			url,
+			method: 'GET',
+		});
+		$('#overlay').hide();
+		return response;
+	}
+
+	/** Change grid height according to screen size */
+	private adjustGridHeight(): void {
+		const gridElement = document.getElementById('grid');
+		const pageCntrl = $('.grid-controls').innerHeight();
+		const screenHeight = $(window).innerHeight();
+		if (gridElement && pageCntrl !== undefined && screenHeight !== undefined) {
+			this.maxGridHeight = screenHeight - pageCntrl;
+			gridElement.style.height = `${this.maxGridHeight}px`;
+		}
+	}
+
+	/** Update the page information and records display based on the current state of the grid. */
+	private updatePageInfo(): void {
+		const totalPages = Math.ceil(this.totalItems / this.pageSize);
+		const pageInfo = `Page ${this.currentPage} of ${totalPages}`;
+		const from = this.firstVal;
+		let to = Math.min(from + this.pageSize, this.maxRange);
+		$('#pageInfo').text(`${pageInfo}`);
+		$('.records').text(`Showing records ${from} to ${to}`);
+	}
+
+	private setupControls(): void {
+		$('#prevBtn').on('click', () => this.handlePageChange(-1));
+		$('#nextBtn').on('click', () => this.handlePageChange(1));
+		$(window).on('resize', debounce(this.handleResize.bind(this), 100));
```

Comment:
We prefer to use arrow functions instead of using `.bind(this)`.
```suggestion
		$(window).on('resize', debounce(() => { this.handleResize(); }, 100));
```